### PR TITLE
Feature/correct log writer in5.6

### LINKED ISSUE #17: Is not working with Laravel 5.6's new logging
5.6 has new logging, which will break this package...

```php
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover

   ReflectionException  : Class Illuminate\Log\Writer does not exist

  at vendor/laravel/framework/src/Illuminate/Container/Container.php:826
    822|
    823|             // If the class is null, it means the dependency is a string or some other
    824|             // primitive type which we can not resolve since it is not a class and
    825|             // we will just bomb out with an error since we have no-where to go.
  > 826|             $results[] = is_null($dependency->getClass())
    827|                             ? $this->resolvePrimitive($dependency)
    828|                             : $this->resolveClass($dependency);
    829|         }
    830|

  Exception trace:

  1   ReflectionParameter::getClass()
      /vendor/laravel/framework/src/Illuminate/Container/Container.php:826

  2   Illuminate\Container\Container::resolveDependencies()
      /vendor/laravel/framework/src/Illuminate/Container/Container.php:795

  Please use the argument -v to see more details.
Script @php artisan package:discover handling the post-autoload-dump event returned with error code 1
```

### PR DESCRIPTION
**Pull request by jimmypuckett**
With Laravel 5.6, the ```Illuminate\Log\Writer``` has been renamed to a new class, but they are both implementations of ```Psr\Log\LoggerInterface```, so change type hint.

Fixes #17

### CODE REVIEW

**Comment by ssfinney** [comment]
@jimmypuckett Merge conflicts

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,6 @@
     "require": {
         "php": ">=5.5.0",
         "illuminate/console": "~5.1.10|5.2.*|5.3.*|5.4.*|5.5.*|5.6.*",
-        "illuminate/log": "~5.1.10|5.2.*|5.3.*|5.4.*|5.5.*|5.6.*",
```
Comment (ssfinney) [inline composer.json]: Don't we still need this, just so that we have a package that implements that LoggerInterface? Or does it not matter?

**Comment by ssfinney** [comment]
Comments are up

**Comment by jimmypuckett** [comment]
@ssfinney I have corrected the merge conflict, so this is ready.  As for dropping the illuminate/log requirement, that is correct as we don't need it as we are type hinting a the ```LoggerInterface```, which is included by other packages...

```bash
$ composer depends psr/log
satooshi/php-coveralls  v1.1.0  requires  psr/log (^1.0)
symfony/debug           v4.0.8  requires  psr/log (~1.0)
```

However, I do think that you are correct that we need t be explicit about needing that class, so I will include it in the ```composer.json```.

Thanks for catching that.

**Comment by jimmypuckett** [comment]
@ssfinney Once this build passes, this is ready to review.  If you merge it, then I will push a release.

**Comment by ssfinney** [comment]
@jimmypuckett Merged, tag please.

**Pull request merged**